tensorcircuit/templates/hamiltonians.py

```diff
@@ -0,0 +1,139 @@
+import typing
+from typing import cast
+from scipy.sparse import coo_matrix
+import numpy as np
+import tensorcircuit as tc
+from .lattice import AbstractLattice
+
+
+def generate_heisenberg_hamiltonian(
+    lattice: AbstractLattice, j_coupling: float = 1.0
```

Comment:
j_coupling is better be a vector whose dimension can be 1, 3, representing for 1: $J_x=J_y=Jz$; 3 different $J_{x, y, z}$